homeassistant/components/raspihats/binary_sensor.py

```diff
@@ -84,7 +97,7 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 class I2CHatBinarySensor(BinarySensorEntity):
     """Representation of a binary sensor that uses a I2C-HAT digital input."""
 
-    I2C_HATS_MANAGER = None
+    I2C_HATS_MANAGER: I2CHatsManager | None = None
 
```

Comment:
The callbacks below should be registered in `async_added_to_hass` not in init method. That's a bug.

Reply:
If you prefer me to make a separate PR I can also revert this change.